docs/proposals/devfile/outer-loop-build-and-deploy.md

```diff
@@ -0,0 +1,438 @@
+
+#### Examples of using `deploy` group command:
+##### Kubernetes deployment manifest:
+    components:
+    - name: myk8sdeploy
+        kubernetes:
+        uri: deploy/deployment-manifest.yaml
+
+    commands:
+    - id: deployk8s
+        apply:
+        component: myk8sdeploy
+        group:
+            kind: deploy
+            isDefault: true
+        attributes: 
+        - name: CONTAINER_IMAGE
+            value: {{myimage}}
+
+`uri`: Kubernetes manifest location (can use `kubectl` to deploy) which can be an URL or a path relative to the devfile. [Example of Kubernetes deployment](#markdown-header-example-of-kubernetes-deployment-manifest) manifest and [example of Operator deployment manifest](#markdown-header-example-of-operator-deployment-manifest).
+
+Variables that need to be replaced during the deployment can be specified using a new `attributes` definition under the `apply` command. One usage example is to pass the image name along to the deployment manifest. A common practice is to use a global variable, e.g. `myimage` in the example above, to refer to the image built in the image built stage.
+ 
+##### Kubernetes deployment manifest (inlined):
+    components:
+    - name: myk8deploy
+        kubernetes:
+        inlined: |
+            apiVersion: batch/v1
+            kind: Job
+            metadata:
+            name: pi
+            spec:
+            template:
+                spec:
+                containers:
+                - name: job
+                    image: {{myimage}}
+                    command: ["some",  "command", "with", "parameters"]
+                restartPolicy: Never
+            backoffLimit: 4
+    commands:
+    - id: deployk8s
+        apply:
+        component: myk8sdeploy
+        group:
+            kind: deploy
+            isDefault: true
+
+##### Helm:
+    components:
+    - name: myhelmdeploy
+        helm:
+        chart: http://helm-chart-url
+        values: 
+            image: quay.io/sample/hello-world # A chart may expose a well-known values.yaml parameter called "image".
+            replicas: 3
+    commands:
+    - id: deployHelm
+        apply:
+        component: myhelmdeploy
+        group:
+            kind: deploy
+            isDefault: true
+        variables: 
+        - name: CONTAINER_IMAGE
+            value: {{myimage}}
+
+
+### Deployment manifest examples
+#### Example of Kubernetes deployment manifest:
+    ---
+    kind: Deployment
+    apiVersion: apps/v1
+    metadata:
+    name: {{.COMPONENT_NAME}}
+    spec:
+    replicas: 1
+    selector:
+        matchLabels:
+        app: {{.COMPONENT_NAME}}
+    template:
+        metadata:
+        creationTimestamp: null
+        labels:
+            app: {{.COMPONENT_NAME}}
+        spec:
+        containers:
```

Comment:
Could you fix the indentation errors throughout -- I don't know if it's a misplaced tab character or something.